### ISSUE TITLE
Puma isn't listening to port option

### ISSUE THREAD

**Issue by blackham** [puma/puma#1217]
### Steps to reproduce
Install puma 3.7.0
start rails with port option (rails server -p 3001 or edit def default_options super.merge({:Port => 3002 ...

### Expected behavior
Puma should be listening on port 3001
This is the output of the previous version, which worked. 
```
rails server -p 3001
=> Booting Puma
=> Rails 5.0.1 application starting in development on http://localhost:3001
=> Run `rails server -h` for more startup options
Puma starting in single mode...
* Version 3.6.2 (ruby 2.3.1-p112), codename: Sleepy Sunday Serenity
* Min threads: 5, max threads: 5
* Environment: development
* Listening on tcp://localhost:3001
```

### Actual behavior
```
rails server -p 3001
=> Booting Puma
=> Rails 5.0.1 application starting in development on http://localhost:3001
=> Run `rails server -h` for more startup options
Puma starting in single mode...
* Version 3.7.0 (ruby 2.3.1-p112), codename: Snowy Sagebrush
* Min threads: 5, max threads: 5
* Environment: development
* Listening on tcp://0.0.0.0:3000
Use Ctrl-C to stop

```

Note Listening on tcp://0.0.0.0:**3000**
I was able to change the port by setting the system environment var PORT. 

```
 env PORT=4444 rails server -p 3001
=> Booting Puma
=> Rails 5.0.1 application starting in development on http://localhost:3001
=> Run `rails server -h` for more startup options
Puma starting in single mode...
* Version 3.7.0 (ruby 2.3.1-p112), codename: Snowy Sagebrush                                                                                                                                            
* Min threads: 5, max threads: 5                                                                                                                                                                        
* Environment: development                                                                                                                                                                              
* Listening on tcp://0.0.0.0:4444                                                                                                                                                                       
Use Ctrl-C to stop   
```

### System configuration
ruby 2.3.1p112 (2016-04-26 revision 54768) [x86_64-linux]
rails 5.0.1

**Comment by IngoAlbers** [comment]
Duplicate of #1212 - will be fixed in 3.7.1

**Comment by nateberkopec** [comment]
Which is being released *right now*!
